Fixes #30943 - adds additional remote db notes (#1696)
theforeman/theforeman.org: plugins/katello/3.16/user_guide/remote_databases/index.md

In either scenario, both of the databases don't have to be remote. You can opt to use only a remote mongo database or only a remote postgresql database. Both postgresql and mongo databases can be on the same host, but this isn't recommended due to the amount of resources mongo can use.

### Important note: Katello 3.15 changes the default database for Pulp from PostgreSQL v9.x to PostgreSQL v12. External databases should be upgraded to avoid conflicts.

### Important note: Katello 3.16 adds dependencies provided by the `rh-postgresql12-postgresql-evr` package. This should be installed on any external PostgreSQL v12 database before applying any db migrations.

> Note that [Forman v2.0 requires PostgreSQL v12](https://theforeman.org/manuals/2.0/index.html#Headlinefeatures).

Reviewer: Wasn't 3.15 already based on Foreman 2.0?
```suggestion
Note that [Foreman v2.0 requires PostgreSQL v12](https://theforeman.org/manuals/2.0/index.html#Headlinefeatures).
```

Edit: Also, Foreman requires 10+, not 12. However, we only build `rh-postgresql12-postgresql-evr` for 12.

In either scenario, both of the databases don't have to be remote. You can opt to use only a remote mongo database or only a remote postgresql database. Both postgresql and mongo databases can be on the same host, but this isn't recommended due to the amount of resources mongo can use.

### Important notes:

> 1. Pulpcore installs a separate PostgreSQL database and lacks support for external databases. Databases should be upgraded to avoid conflicts.

Reviewer: Version 3.16 does support external databases
```suggestion
```

Reviewer: This was marked as resolved, but 3.16 does support a separate PG DB for Pulpcore. If you keep a list, I'd suggest not using numbers but rather bullets.

> 2. Katello 3.16 adds dependencies provided by the `rh-postgresql12-postgresql-evr` package. This should be installed on any external PostgreSQL v12 database before applying any db migrations.

Reviewer: ```suggestion
2. Katello 3.16 adds a dependency on [postgresql-evr](https://github.com/katello/postgresql-evr). The Katello repository provides the `rh-postgresql12-postgresql-evr` package on EL7. This should be installed on any external PostgreSQL v12 database before applying any db migrations.
```

> Note that [Foreman v2.0 requires PostgreSQL v10+](https://theforeman.org/manuals/2.0/index.html#Headlinefeatures).

Reviewer: Katello 3.16 runs on Foreman 2.1. This note belongs in the 3.15 docs.

> Note that [Foreman v2.1 requires PostgreSQL v10+](https://theforeman.org/manuals/2.1/index.html#Headlinefeatures).

Reviewer: It was part of version 2.0, but it's not a headline feature in 2.1. The link should be removed but I'd also consider dropping it altogether. The evr package is only built for PG 12 so that already narrows it down for most users.

Reviewer: The link is still incorrect.
